javascript/ql/src/Expressions/MissingReturnValueGetsUsed.ql

```diff
@@ -0,0 +1,88 @@
+/**
+ * @name Missing Return Value Gets Used
+ * @description A call to a function that doesn't return a value is used in a context where the
+ *              value of the call actually matters.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/missing-return-value-gets-used
+ * @precision very-high
+ * @tag correctness
+ */
+
+import javascript
+import semmle.javascript.CFG
+ 
+/*
+ * A function with no return value can be called without being problematic under the following
```

Comment:
How do you mean?

Should I elaborate for each case, providing explanations for why?

Reply:
Yes, please. Not in the code, I'd just like to know why you don't consider those cases problematic.

Reply:
It might still be good to have it in the comment, b/c if you're wondering, others will wonder too. I also noticed that there's at least one explanation for valid cases later on, so I can' use the same structure. Generally:

```
A function with no return value can be called without being problematic under the following conditions on the internal properties of the function:

- When the function is completely empty and has no statements in it, because this is often done as a form of stubbing in order to provide something that indicates the bound variable is intended to be for a function
- When the function is in an error function (TODO)
```

For error functions, I want to confirm with folx first about the intent of the description that is used in deepscan. They say "Error function, which always throws exception". Do you think this means a specific kind of function, or just any old function which happens to only ever throw exceptions as final execution nodes?

Reply:
Oh, do you mean you added this special case to mirror what deepscan does? I don't think that's a good idea. We should be able to do better than them. In particular, an error function that always throws an exception should already not be flagged due to us reasoning about control flow. The condition about empty functions sounds fishy to me. It seems like a half-baked way of avoiding false positives.

Reply:
It does indeed sound half-baked. Tho a lot of big libraries do this. Like, it's extremely common in Vue, for instance. I'm morally opposed to this kind of weird stubbing, but it's also a behavior that modern frontend devs seem to be really into? I'm unsure what would be a good change here.